web/src/app/admin/admin-api-keys/AdminAPIKeyExpirationField.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react'
+import { FormContainer } from '../../forms'
+import InputLabel from '@mui/material/InputLabel'
+import MenuItem from '@mui/material/MenuItem'
+import FormControl from '@mui/material/FormControl'
+import Select, { SelectChangeEvent } from '@mui/material/Select'
+import Grid from '@mui/material/Grid'
+import { Typography } from '@mui/material'
+import makeStyles from '@mui/styles/makeStyles'
+import { ISODatePicker } from '../../util/ISOPickers'
+import { DateTime } from 'luxon'
+
+const useStyles = makeStyles(() => ({
+  expiresCon: {
+    'padding-top': '15px',
+  },
+}))
+// props object for this compoenent
+interface FieldProps {
+  setValue: (val: string) => void
+  value: string
+  create: boolean
+}
+
+export default function AdminAPIKeyExpirationField(
+  props: FieldProps,
+): JSX.Element {
+  const classes = useStyles()
+  const { value, setValue } = props
+  const [dateVal, setDateVal] = useState<string>(value)
+  const [options, setOptions] = useState('7')
+  const [showPicker, setShowPicker] = useState(false)
+  // handles expiration date field options changes: sets and computes expirates at value based on the selected additional days value or sets value to +7 days today if custom option is selected
+  const handleChange = (event: SelectChangeEvent): void => {
+    const val = event.target.value as string
+    setOptions(val)
+    setShowPicker(val.toString() === '0')
+
+    if (val !== '0') {
+      setDateVal(
+        DateTime.now()
+          .plus({ days: parseInt(val) })
+          .toLocaleString({
+            weekday: 'short',
+            month: 'short',
+            day: '2-digit',
+            year: 'numeric',
+            hour: '2-digit',
+            minute: '2-digit',
+            second: '2-digit',
+          }),
+      )
+    }
+  }
+  // handles custon expiration date field option changes: sets and computes expires at value based on the selected date
+  const handleDatePickerChange = (val: string): void => {
+    // eslint-disable-next-line prettier/prettier
+    if (val != null) {
+      setDateVal(
+        new Date(val).toLocaleString([], {
+          weekday: 'short',
+          month: 'short',
+          day: '2-digit',
+          year: 'numeric',
+          hour: '2-digit',
+          minute: '2-digit',
+          second: '2-digit',
+        }),
+      )
+    }
+  }
+  // set expirate at date value to state
+  useEffect(() => {
+    setValue(dateVal)
+  })
```

Comment:
Not sure why this is necessary -- it will cause an immediate re-render on mount, but `dateVal` should already be `props.value` so I don't think it's needed.